pycbc/waveform/ringdown.py

```diff
@@ -501,20 +738,36 @@ def multimode_base(input_params, domain, freq_tau_approximant=False):
         for mode in taus:
             if 'delta_tau{}'.format(mode) in input_params:
                 taus[mode] += input_params['delta_tau{}'.format(mode)] * taus[mode]
-
     if domain == 'td':
         outplus, outcross = td_output_vector(freqs, taus,
                             input_params['taper'], input_params['delta_t'],
                             input_params['t_final'])
         for lmn in freqs:
-            hplus, hcross = td_damped_sinusoid(freqs[lmn], taus[lmn],
-                            amps[lmn], phis[lmn], outplus.delta_t,
-                            outplus.sample_times[-1], int(lmn[0]), int(lmn[1]),
-                            input_params['inclination'])
+            #hplus, hcross = td_damped_sinusoid(freqs[lmn], taus[lmn],
+            #                amps[lmn], phis[lmn], outplus.delta_t,
+            #                outplus.sample_times[-1], int(lmn[0]), int(lmn[1]),
+            #                input_params['inclination'], pols[lmn])
+            if harmonics == 'spheroidal':
+                final_spin = input_params['final_spin']
+            else:
+                final_spin = None
+            if amps[lmn] == 0.:
+                # skip
+                continue
+            hplus, hcross = td_damped_sinusoid(
+                freqs[lmn], taus[lmn], amps[lmn], phis[lmn],
+                outplus.delta_t, outplus.sample_times[-1],
+                l=int(lmn[0]), m=int(lmn[1]), n=int(lmn[2]),
+                inclination=input_params['inclination'],
+                azimuthal=input_params['azimuthal'],
+                dphi=dphis[lmn], dbeta=dbetas[lmn],
+                harmonics=harmonics, final_spin=final_spin,
+                pol=pols[lmn], polnm=polnms[lmn])
             if input_params['taper'] and outplus.delta_t < taus[lmn]:
                 taper_hp, taper_hc = apply_taper(taus[lmn], amps[lmn],
                                      phis[lmn], outplus.delta_t, int(lmn[0]),
-                                     int(lmn[1]), input_params['inclination'])
+                                     int(lmn[1]), input_params['inclination'],
+                                     pols[lmn])
                 t0 = -int(outplus.start_time * outplus.sample_rate)
                 outplus[t0-len(taper_hp):t0].data += taper_hp
                 outplus[t0:].data += hplus
```

Comment:
Update the frequency domain version so that it works with the new arguments.

Reply:
Done, although figuring out how to get non-circular polarization (via the dphi and dbeta parameters) was more work than I wanted to do right now, so the frequency domain version doesn't support that. This is the same functionality as before, with the added ability to use spheroidal or arbitrary harmonics.